src/read_distribution.rs

```diff
@@ -125,14 +125,20 @@ fn read_metadata_for_wheel(path: impl AsRef<Path>) -> Result<Vec<(String, String
 /// Only parses the filename since dist-info is not part of source
 /// distributions
 fn read_metadata_for_source_distribution(path: impl AsRef<Path>) -> Result<Vec<(String, String)>> {
+    // "dist/foo_ext-1.0.1.tar.gz" -> "foo_ext-1.0.1/PKG-INFO"
+    let mut pkginfo: PathBuf = path.as_ref().file_name().unwrap().into();
+    pkginfo.set_extension("");
+    pkginfo.set_extension("");
```

Comment:
Duplicated?

Reply:
I'm stripping ".tar.gz" this way. Not pretty, I know.